### PR TITLE
CDAP-13266 Field Level Lineage plugin API.

### PR DESCRIPTION
**Pull request by sagarkapare**
JIRA: https://issues.cask.co/browse/CDAP-13266

Operations are similar to the one exposed in the platform API #10008. However difference is all the operations will take the plain String rather than InputField as an input, since the uniqueness will be guaranteed by the DataPipelineApp by prefixing the stage name to the operation name. 

Also these classes will be extended in future by adding methods which are responsible for propagating metadata in the pipeline from source to targets.

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/lineage/field/ReadOperation.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Represents a read operation from a data source into a collection of output fields.
+ */
+public class ReadOperation extends Operation {
+  private final EndPoint source;
+  private final List<String> outputs;
```
Comment (anew) [inline cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/lineage/field/ReadOperation.java]: I would call these "output fields" (outputFields) and in the Java docs always refer to them as output fields. Just outputs is too generic - what is an output?

**Review by anew — COMMENTED** [review]
Only a minor comment about naming and javadocs. Other than that LGTM

**Comment by sagarkapare** [comment]
Thanks @anew for the review. Addressed comments will merge once the build is green - https://builds.cask.co/browse/CDAP-DUT6384-3

**Comment by sagarkapare** [comment]
Build is green. Squashing and merging.

**Pull request merged**